src/renderer/services/plugin-manager.js

```diff
@@ -67,11 +68,11 @@ class PluginManager {
       profileId
     })
 
-    await this.loadComponents(pluginObject, plugin)
-    await this.loadRoutes(pluginObject, plugin)
-    await this.loadMenuItems(pluginObject, plugin, profileId)
-    await this.loadAvatars(pluginObject, plugin, profileId)
-    await this.loadWalletTabs(pluginObject, plugin, profileId)
+    // COMPONENTS, ROUTES, MENU_ITEMS, AVATARS, WALLET_TABS
+    for (const permission of uniq(plugin.config.permissions)) {
```

Comment:
I think it needs a validation before calling the method:

```suggestion
    for (const permission of ['COMPONENTS, ROUTES, MENU_ITEMS, AVATARS, WALLET_TABS']) {
    	if (plugin.config.permission[permission]) {
    		// ...
    	}
    }
```

Reply:
In this case, `permissions` is an Array, so it wouldn't be necessary to do that. The only possible problem that it could cause it's that some permissions should be loaded in order. I'll change it now.

Reply:
You are considering that all plugins will write the permissions correctly.

Reply:
It's an array of permissions, if someone uses numbers, `null` or an object, it wouldn't work and it would throw an Error, so they could realize that it's a mistake.